# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/VideoOutput.h
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 dom/media/VideoOutput.h
--- a/dom/media/VideoOutput.h	Sun Sep 15 20:03:40 2024 +0000
+++ b/dom/media/VideoOutput.h	Sun Sep 15 21:49:38 2024 +0000
@@ -100,8 +100,11 @@
         // We ignore null images.
         continue;
       }
-      images.AppendElement(ImageContainer::NonOwningImage(
-          image, chunk.mTimeStamp, frameId, mProducerID));
+      ImageContainer::NonOwningImage nonOwningImage(
+          image, chunk.mTimeStamp, frameId, mProducerID,
+          chunk.mProcessingDuration, chunk.mMediaTime, chunk.mWebrtcCaptureTime,
+          chunk.mWebrtcReceiveTime, chunk.mRtpTimestamp);
+      images.AppendElement(std::move(nonOwningImage));
 
       lastPrincipalHandle = chunk.GetPrincipalHandle();
 
@@ -281,10 +284,7 @@
 
         // Pick the first frame and run it through the rendering code.
         VideoSegment segment;
-        segment.AppendFrame(do_AddRef(c->mFrame.GetImage()),
-                            c->mFrame.GetIntrinsicSize(),
-                            c->mFrame.GetPrincipalHandle(),
-                            c->mFrame.GetForceBlack(), c->mTimeStamp);
+        segment.AppendFrame(*c);
         VideoOutput::NotifyRealtimeTrackData(aGraph, aTrackOffset, segment);
         return;
       }